### PR TITLE
Select whether to install nimble or 3par related deployment

### PR DESCRIPTION
**Pull request by papanito**

### CODE REVIEW

**Comment by papanito** [comment]
@raunakkumar @rgcostea @suneeth51 can you have a look at my pr. thanks a lot

**Comment by Baykonur** [comment]
Without knowing about this PR, couple of days I ago I created a very similar [PR](https://github.com/hpe-storage/co-deployments/pull/225) which basically does the same thing.

The fact that the original helm chart cannot be configured not to install one of the CSP backends is a limitation in my opinion. Who is using nimble and 3par-primera storage devices at the same time?

It's been 20 days since this one got created but there is not even a single feedback. 
Dear @raunakkumar, @rgcostea, @suneeth51 , could you please review if not advise?

**Review by suneeth51 — APPROVED** [review]
lgtm

**Comment by suneeth51** [comment]
> Without knowing about this PR, couple of days I ago I created a very similar [PR](https://github.com/hpe-storage/co-deployments/pull/225) which basically does the same thing.
> 
> The fact that the original helm chart cannot be configured not to install one of the CSP backends is a limitation in my opinion. Who is using nimble and 3par-primera storage devices at the same time?
> 
> It's been 20 days since this one got created but there is not even a single feedback.
> Dear @raunakkumar, @rgcostea, @suneeth51 , could you please review if not advise?

The changes looks good to me. @raunakkumar Please provide your feedback on this?

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:36]: Remove this. This README is rendered in far too many places that might not interpret MD properly.

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:50]: Remove

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:57]: remove

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:73]: remove

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:83]: remove

File: helm/charts/hpe-csi-driver/templates/nimble-csp.yaml
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/templates/nimble-csp.yaml:1]: Reflect desired changes.

File: helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml:1]: Reflect desired changes.

File: helm/charts/hpe-csi-driver/templates/nimble-csp.yaml
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/templates/nimble-csp.yaml:65]: Insert line breaks. The linting breaks otherwise.

File: helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml:67]: Link break.

File: helm/charts/hpe-csi-driver/values.yaml
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/values.yaml:29]: Reflect changes.

**Review by datamattsson — CHANGES_REQUESTED** [review]
We need more eyes on this before merge. This is a PM decision to merge or not.

**Comment by c-snell** [comment]
This change is a PM call. There was a decision made early on to deploy the CSI Driver with all supported CSPs similar to Trident model. If we are breaking that model, some scenarios to consider:
- How would one revert the change if the CSI Driver is already deployed and re-enable additional backends as the CSI Driver expands supported platforms? 
- How does the helm chart handle upgrades with disabled CSPs? Are there any validations that would break?

I feel this requires more careful consideration before merging this PR.

File: helm/charts/hpe-csi-driver/templates/nimble-csp.yaml
Comment (papanito) [reply helm/charts/hpe-csi-driver/templates/nimble-csp.yaml:65]: will fix

File: helm/charts/hpe-csi-driver/templates/nimble-csp.yaml
Comment (papanito) [reply helm/charts/hpe-csi-driver/templates/nimble-csp.yaml:1]: What you mean with "reflect" what shall I do here?

File: helm/charts/hpe-csi-driver/README.md
Comment (papanito) [reply helm/charts/hpe-csi-driver/README.md:73]: will fix

File: helm/charts/hpe-csi-driver/README.md
Comment (papanito) [reply helm/charts/hpe-csi-driver/README.md:57]: will fix

File: helm/charts/hpe-csi-driver/README.md
Comment (papanito) [reply helm/charts/hpe-csi-driver/README.md:50]: will fix

File: helm/charts/hpe-csi-driver/README.md
Comment (papanito) [reply helm/charts/hpe-csi-driver/README.md:36]: will fix

File: helm/charts/hpe-csi-driver/Chart.yaml
Comment (papanito) [reply helm/charts/hpe-csi-driver/Chart.yaml:12]: Ok will revert change

File: helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml
Comment (papanito) [reply helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml]: Will fix

**Comment by Baykonur** [comment]
> This change is a PM call. There was a decision made early on to deploy the CSI Driver with all supported CSPs similar to Trident model. If we are breaking that model, some scenarios to consider:
> 
> * How would one revert the change if the CSI Driver is already deployed and re-enable additional backends as the CSI Driver expands supported platforms?
> * How does the helm chart handle upgrades with disabled CSPs? Are there any validations that would break?
> 
> I feel this requires more careful consideration before merging this PR.

@c-snell , any indication when HPE can discuss this internally and decide the fate of this PR?

I understand there was a decision about deploying all the CSP backend pods with the helm chart but in reality very few will have the luxury to have all different HPE hardware serving volumes to their internal K8s clusters. 

We are happy that helm chart is ready and complete but because we do not want to deploy nimble-csp resource we will need to copy-paste and tweak the existing helm chart and copying resources from your GitHub repo to our internal repo is not ideal whereas we can only declare it as a dependency if you make it possible to be able to switch-off different CSP resources.

File: helm/charts/hpe-csi-driver/templates/nimble-csp.yaml
Comment (datamattsson) [reply helm/charts/hpe-csi-driver/templates/nimble-csp.yaml:1]: Somehow my first review comment was lost. We need the parameters changed and verbiage updated. Those changes needs to be reflected in the respective template.

|Param|Desc|Default|
| - | - | -|
| disable.nimble | Do not apply Nimble CSP `Service` and `Deployment` | false |
| disable.primera | Do not apply Primera CSP `Service` and `Deployment` | false |

Reversing enable to disable is simply to be in line with `disableNodeConformance` where you need to flag something "true" to change a non-default behaviour.

**Comment by c-snell** [comment]
@Baykonur I wanted to update you on this. This was discussed internally yesterday and we have turned it over to engineering to review and test various installation, upgrade, modification scenarios before a final call will be made. We don't have an ETA at the moment when this testing will be complete. Also once it is approved it would go into the next release of the CSI Driver so best case scenario the ability to choose which CSP is installed through the Helm chart would come in the 1.5 release.

**Pull request closed without merging**
